### PR TITLE
Video usb logger

### PR DESCRIPTION
**Pull request by dewagter**
Re-use view-video to log entire stream with attitude and position

### CODE REVIEW

**Comment by flixr** [comment]
Um, I guess you meant to create the video_usb_logger ontop of master and not v5.4?

**Comment by dewagter** [comment]
rebased on master. (developed on stable flying code)

File: sw/airborne/modules/computer_vision/viewvideo.c [hunk 1]
```diff
@@ -206,7 +224,7 @@ void *computervision_thread_main(void *data)
       0,                // Format 422
       quality_factor,   // Jpeg-Quality
       dri_jpeg_header,  // DRI Header
-      1                 // 90kHz time increment
+      0                 // 90kHz time increment
```
Comment (dewagter) [inline sw/airborne/modules/computer_vision/viewvideo.c]: Use linux time stamp (works at any frame rate).

File: sw/airborne/modules/computer_vision/viewvideo.c [hunk 2]
```diff
@@ -148,11 +155,11 @@ void *computervision_thread_main(void *data)
...
     // compute usleep to have a more stable frame rate
-    struct timeval time;
-    gettimeofday(&time, NULL);
-    int dt = (int)(time.tv_sec - last_time.tv_sec) * 1000000 + (int)(time.tv_usec - last_time.tv_usec);
+    struct timeval vision_thread_sleep_time;
+    gettimeofday(&vision_thread_sleep_time, NULL);
+    int dt = (int)(vision_thread_sleep_time.tv_sec - last_time.tv_sec) * 1000000 + (int)(vision_thread_sleep_time.tv_usec - last_time.tv_usec);
     if (dt < microsleep) { usleep(microsleep - dt); }
-    last_time = time;
+    last_time = vision_thread_sleep_time;
```
Comment (dewagter) [inline sw/airborne/modules/computer_vision/viewvideo.c]: compile warnings (shadows global time)

File: sw/airborne/modules/computer_vision/viewvideo.c [hunk 3]
```diff
@@ -181,11 +198,12 @@ void *computervision_thread_main(void *data)
             // leave for loop
             break;
           } else {
-            printf("file exists\n");
+            //printf("file exists\n");
```
Comment (dewagter) [inline sw/airborne/modules/computer_vision/viewvideo.c]: Lot of overhead on large image numbers

File: sw/airborne/modules/computer_vision/viewvideo.c [hunk 4]
```diff
@@ -181,11 +198,12 @@ void *computervision_thread_main(void *data)
...
           }
         }
       }
-      computer_vision_thread_command = 1;
+      if (computer_vision_thread_command == 2)
+        computer_vision_thread_command = 1;
```
Comment (dewagter) [inline sw/airborne/modules/computer_vision/viewvideo.c]: this variable is not thread safe anymore: before only the module could instruct the thread to close, now the thread also writes this variable so it is possible that both write it at the same time. This change is not a fix, but reduces the critical time from jpeg-compression time to a single if statement.

File: conf/modules/video_usb_logger.xml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE module SYSTEM "module.dtd">
+
+<module name="computer_vision">
```
Comment (flixr) [inline conf/modules/video_usb_logger.xml]: name should be video_usb_logger and dir computer_vision

File: sw/airborne/modules/computer_vision/video_usb_logger.c [hunk 6]
```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2015
```
Comment (flixr) [inline sw/airborne/modules/computer_vision/video_usb_logger.c]: copyright of who?

File: sw/airborne/modules/computer_vision/video_usb_logger.c [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+ */
+
+#include "video_usb_logger.h"
+
+#include <stdio.h>
+#include "state.h"
+#include "viewvideo.h"
+
+/** Set the default File logger path to the USB drive */
+#ifndef FILE_LOGGER_PATH
```
Comment (flixr) [inline sw/airborne/modules/computer_vision/video_usb_logger.c]: should also check for VIDEO_USB_LOGGER_PATH ?

File: conf/modules/video_usb_logger.xml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  <doc>
+    <description>Logs attitude and position to a csv and images to jpeg files (only for linux)</description>
```
Comment (flixr) [inline conf/modules/video_usb_logger.xml]: Please remember that first line of the description (until dot) is used as name/title for the module in the generated dox.

File: sw/airborne/modules/computer_vision/video_usb_logger.c [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+  }
+}
+
+/** Log the values to a csv file */
+void video_usb_logger_periodic(void)
+{
+  if (video_usb_logger == NULL) {
+    return;
+  }
+  static uint32_t counter;
```
Comment (flixr) [inline sw/airborne/modules/computer_vision/video_usb_logger.c]: init with 0

File: sw/airborne/modules/computer_vision/viewvideo.c [hunk 10]
```diff
@@ -100,6 +102,11 @@ void *computervision_thread_main(void *data)
   vid.device = (char *)"/dev/video1";
   vid.w = 1280;
   vid.h = 720;
+//#if BOTTOM_CAMERA == 1
```
Comment (flixr) [inline sw/airborne/modules/computer_vision/viewvideo.c]: check is commented out?
Also would probably be nicer `#if USE_BOTTOM_CAMERA` or `#if VIEW_BOTTOM_CAMERA`

**Comment by dewagter** [comment]
Ready

**Pull request closed without merging**